crates/pathfinder/src/state/sync.rs

```diff
@@ -471,72 +411,27 @@ async fn update_sync_status_latest(
         let transaction = connection
             .transaction_with_behavior(TransactionBehavior::Immediate)
             .context("Create database transaction")?;
 
-        for update in updates {
-            L1StateTable::upsert(&transaction, update).context("Insert update")?;
-        }
-
-        // Track combined L1 and L2 state.
-        let l1_l2_head = RefsTable::get_l1_l2_head(&transaction).context("Query L1-L2 head")?;
-        let expected_next = l1_l2_head
-            .map(|head| head + 1)
-            .unwrap_or(BlockNumber::GENESIS);
-
-        match updates.first() {
-            Some(update) if update.block_number == expected_next => {
-                let mut next_head = None;
-                for update in updates {
-                    let l2_root = StarknetBlocksTable::get_state_commitment(
-                        &transaction,
-                        update.block_number.into(),
-                    )
-                    .context("Query L2 root")?
-                    .map(|(a, b)| StateCommitment::calculate(a, b));
-
-                    match l2_root {
-                        Some(l2_root) if l2_root == update.global_root => {
-                            next_head = Some(update.block_number);
-                        }
-                        _ => break,
-                    }
-                }
-
-                if let Some(next_head) = next_head {
-                    RefsTable::set_l1_l2_head(&transaction, Some(next_head))
-                        .context("Update L1-L2 head")?;
-                }
-            }
-            _ => {}
-        }
-
-        transaction.commit().context("Commit database transaction")
-    })
-}
-
-async fn l1_reorg(connection: &mut Connection, reorg_tail: BlockNumber) -> anyhow::Result<()> {
-    tokio::task::block_in_place(move || {
-        let transaction = connection
-            .transaction_with_behavior(TransactionBehavior::Immediate)
-            .context("Create database transaction")?;
+        L1StateTable::upsert(&transaction, update).context("Insert update")?;
 
-        L1StateTable::reorg(&transaction, reorg_tail).context("Delete L1 state from database")?;
+        let l2_root =
+            StarknetBlocksTable::get_state_commitment(&transaction, update.block_number.into())
+                .context("Query L2 root")?
+                .map(|(a, b)| StateCommitment::calculate(a, b));
 
-        // Track combined L1 and L2 state.
-        let l1_l2_head = RefsTable::get_l1_l2_head(&transaction).context("Query L1-L2 head")?;
-        match l1_l2_head {
-            Some(head) if head >= reorg_tail => {
-                let new_head = match reorg_tail {
-                    BlockNumber::GENESIS => None,
-                    other => Some(other - 1),
-                };
-                RefsTable::set_l1_l2_head(&transaction, new_head).context("Update L1-L2 head")?;
+        if let Some(l2_root) = l2_root {
+            if l2_root == update.global_root {
```

Comment:
This keeps the existing semantics of using the state commitment as the point of comparison. We should switch this to using the block hash instead as it is a stronger guarantee (since it includes the state commitment). But we can do this in a separate PR if you want.

Reply:
We should also definitely log if `L1 != L2` probably as `ERROR` or at the very least `WARN`. 

In addition, we might want to consider attempting a repair i.e. finding the last point at which `L1 == L2` still holds true, by searching backwards through the `L1 state table`.

Reply:
Added a hash match check and a `WARN`. If hashes do not match (even though "it should never happen" (c)), then the most recent block where hashes did match must be in refs table, as l1_state and refs are updated simultaneously.